src/platform_impl/ios/uikit/status_bar.rs

```diff
@@ -0,0 +1,16 @@
+use icrate::Foundation::NSInteger;
+use objc2::encode::{Encode, Encoding};
+
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+#[allow(dead_code)]
+#[repr(isize)]
+pub enum UIStatusBarStyle {
+    #[default]
+    Default = 0,
+    LightContent = 1,
+    DarkContent = 3,
+}
```

Comment:
Could we just use `with_theme` that we have, it's exactly the same `light`/`dark` thing?

Reply:
Agreed

Reply:
I tried following this suggestion, but as I'm implementing it, I'm starting to think it may not be a good idea after all. The thing is, the status bar style doesn't define a *theme*, it merely defines the color of the text and the icons in the status bar. The actual theme, including what is displayed in the background of the status bar, is still determined fully by the app developer. But if they use a light background (what you may call a "light theme"), they should choose `DarkContent`, because it needs to contrast. And similarly "dark themes" require using light content.

Of course I could just reverse the meaning in code, so that `Theme::Dark` maps to `LightContent` and `Theme::Light` to `DarkContent`, but it feels confusing at best.

Alternatively, I could also *not* revert the meaning and specify that `Theme::Light` means light content, but that's confusing too since I feel it would invert the meaning of "theme" compared to other platforms.

So given that there's already the iOS-specific `set_prefers_status_bar_hidden()` I'm now thinking it's better to just leave `set_preferred_status_bar_style()` right next to it, so they're easier to find and use together, and hopefully it'll be clearer to just use the iOS-specific terminology, since app developers usually need to configure it elsewhere in their `.plist` as well, where the same terminology is used.

Reply:
theme defines a theme not controlled by your app directly. On macOS it's the style of the header bar.

Reply:
Right, but that means the situation is not really comparable to iOS, right? On iOS the background of the status bar is controlled directly by the app, while `set_preferred_status_bar_style()` controls only the foreground color of the text and icons in the status bar.

Reply:
We could probably somehow make it clear with the docs? For example, when you set a `with_theme(Dark)` it'll be white text, and the other way black text. Like I really don't want to have more options, and in general applications do match the content. If they want to make a contrast, they surely can achieve so.

Just tell in the docs explicitly what this variable is actually doing.

Reply:
Actually, I think `with_theme` should be using `overrideUserInterfaceStyle`, what I was agreeing to was to reuse the `Theme` struct.

Could someone with a newer iOS device than I check if setting that also changes the status bar style? If not, then I believe the current approach (add a specific option to control the preferred status bar style) is fine.

Reply:
I agree it would make sense if `with_theme()` was implemented to use `overrideUserInterfaceStyle()`. There the `Theme` struct makes sense indeed.

But regardless of whether the theme affects the *default* status bar look (which it doesn't appear to always do, see for instance: https://stackoverflow.com/questions/70518085/how-to-change-status-bar-colour-whenever-i-overrideuserinterfacestyle), it wouldn't be sufficient for all situations. For instance, imagine an app that uses the light theme, but displays a dark background in the upper part of the screen. Such an app would still want to explicitly override the status bar style, so I see the two functions more as orthogonal to one another.